kse/src/main/java/org/kse/crypto/publickey/JwkPublicKeyExporter.java

```diff
@@ -0,0 +1,141 @@
+package org.kse.crypto.publickey;
+
+import java.nio.charset.StandardCharsets;
+import java.security.PublicKey;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.ECParameterSpec;
+import java.util.Map;
+import java.util.Optional;
+
+import org.bouncycastle.jce.spec.ECNamedCurveSpec;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.RSAKey;
+
+public class JwkPublicKeyExporter {
+    private final String alias;
+    private final KeyExporter keyExporter;
+
+    private JwkPublicKeyExporter(PublicKey publicKey, String alias) {
+        this.alias = alias;
+        if (publicKey instanceof ECPublicKey) {
```

Comment:
You are right, I wish I had started with proper tests for each public key type, instead of only for RSA and ECC. Thank you for pointing this problem out.
Problem is solved now - I have added specialized instance of `JwkPublicKeyExporter.KeyExporter` that handles Edwards curves